documentation/src/main/asciidoc/user_guide/streams.adoc

```diff
@@ -118,18 +118,16 @@ This relies on the spec to pick the most specific method as defined link:https:/
 
 In our previous example we used a `Collector` to collect all the results into a `Map`.
 Unfortunately the link:{jdkdocroot}/java/util/stream/Collectors.html[Collectors]
-class doesn't produce Serializable instances.  Thus if you need to use these, you can use the newly provided
-link:{javadocroot}/org/infinispan/stream/CacheCollectors.html[CacheCollectors]
-class which allows for a `Supplier<Collector>` to be provided.  This instance could then use the
-link:{jdkdocroot}/java/util/stream/Collectors.html[Collectors]
-to supply a `Collector` which is not serialized. You can read more details about how the
-collector peforms in a distributed fashion at link:user_guide.html#distributed_stream_execution[distributed execution].
+class doesn't produce Serializable instances.
+Thus if you need to use these, you should call the overloaded `collect`methods that take `Supplier<Collector>`.
+This can be serialized enabling the collector to be lazily instantiated in remote nodes.
+You can read more details about how the collector peforms in a distributed fashion at link:user_guide.html#distributed_stream_execution[distributed execution].
```

Comment:
We should probably leave in a blurb about the `CacheCollectors` still. This is useful if your is defined as a `Stream` interface which doesn't have the new collect method taking a Supplier.

Reply:
Not sure what you exactly want... mind sending a PR to my branch?